### PR TITLE
bpo-40903: Handle multiple '=' in invalid assignment rules in the PEG parser

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue40903

Automerge-Triggered-By: @pablogsal

### CODE REVIEW

File: Grammar/python.gram [hunk 1]
```diff
@@ -646,7 +646,7 @@ invalid_assignment:
         RAISE_SYNTAX_ERROR_KNOWN_LOCATION(a, "only single target (not tuple) can be annotated") }
     | a=expression ':' expression ['=' annotated_rhs] {
         RAISE_SYNTAX_ERROR_KNOWN_LOCATION(a, "illegal target for annotation") }
-    | a=star_expressions '=' (yield_expr | star_expressions) {
+    | a=(z=(star_expressions) '=' {z})+ (yield_expr | star_expressions) {
```
Comment (gvanrossum) [inline Grammar/python.gram]: Hm, `z=(star_expressions)` has redundant parentheses. Probably the `)` should be after `'='`?

Comment (pablogsal) [reply Grammar/python.gram]: Yep, I have also pushed a new commit to handle mixed cases like `x = y = z = True = 3` in 06ecf9d

**Comment by stestagg** [comment]
Hi

I think I've found another one:

`a=yield=1` 

seems to segfault even with these patches

Steve

**Comment by pablogsal** [comment]
> I think I've found another one:

Thanks for the catch! Handled in ac498b4

**Review by gvanrossum — APPROVED** [review]
LGTM

**Pull request merged**